pkgs/development/interpreters/spidermonkey/91.nix

```diff
@@ -0,0 +1,119 @@
+{ lib, stdenv
+, fetchurl
+, pkg-config
+, perl
+, python3
+, zip
+, buildPackages
+, which
+, readline
+, zlib
+, icu69
+, cargo
+, rustc
+, rust-cbindgen
+, yasm
+, llvmPackages_latest
+, nspr
+, m4
+}:
+
+stdenv.mkDerivation rec {
+  pname = "spidermonkey";
+  version = "91.2.0";
+
+  src = fetchurl {
+    url = "mirror://mozilla/firefox/releases/${version}esr/source/firefox-${version}esr.source.tar.xz";
+    sha256 = "1hs2bvzl0d4kfir3gq997kwxm90ygapqn6xlw47cihnh479wzwry";
+  };
+
+  outputs = [ "out" "dev" ];
+  setOutputFlags = false; # Configure script only understands --includedir
+
+  nativeBuildInputs = [
+    cargo
+    llvmPackages_latest.llvm # for llvm-objdump
+    perl
+    pkg-config
+    python3
+    rust-cbindgen
+    rustc
+    which
+    yasm # to buid icu? seems weird
+    zip
+    m4
+  ];
+
+  buildInputs = [
+    icu69
+    nspr
+    readline
+    zlib
+  ];
+
+  preConfigure = ''
+    export LIBXUL_DIST=$out
+    export PYTHON="${buildPackages.python3.interpreter}"
+    export M4=m4
+    export AWK=awk
+    export AC_MACRODIR=$PWD/build/autoconf/
+
+    pushd js/src
+    sh ../../build/autoconf/autoconf.sh --localdir=$PWD configure.in > configure
```

Comment:
Should we use bash explicitly?

Reply:
Can you discuss your train of thought?

Reply:
sh can be either dash, ash or bash depending where you are and it would be nice to be explicit about this. For NixOS IIRC /bin/sh is different in the sandbox and outside the sandbox.

Reply:
@lostnet Your decision as a maintainer.

Reply:
Looking at patchShebangs, I think it would take the first sh on the $path and it is also relying on this normally being bash's sh from the stdenv in a mkDerivation.

If that looks correct and no one thinks patchShebangs is going to be changed this seems like bug for bug compatible emulation and I'd leave it as is. I think fixing the script up to then call patchShebangs on it is the other alternative that guarantees normal even if normal changes.